pkg/util/transport_builder.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2018 Google, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"time"
+
+	. "github.com/google/go-containerregistry/pkg/name"
+)
+
+var tlsConfiguration = struct {
+	registriesCertificates map[string]string
+	skipTsVerifyRegistries map[string]struct{}
+}{
+	registriesCertificates: make(map[string]string),
+	skipTsVerifyRegistries: make(map[string]struct{}),
+}
+
+func ConfigureTls(skipTsVerifyRegistries []string, registriesToCertificates map[string]string) {
+	tlsConfiguration.registriesCertificates = make(map[string]string)
```

Comment:
nit: I would move this line down above the for loop where you initialize the map values